examples/tv-app/linux/include/wake-on-lan/WakeOnLanManager.cpp

```diff
@@ -19,24 +19,37 @@
 #include "WakeOnLanManager.h"
 #include <fstream>
 #include <iostream>
+#include <lib/support/BytesToHex.h>
+#include <platform/ConfigurationManager.h>
 #include <string>
 
 using namespace chip;
 using namespace chip::app::Clusters::WakeOnLan;
 
 std::string getMacAddress()
 {
-    std::ifstream input("/sys/class/net/eth0/address");
-    std::string line;
-    std::getline(input, line);
-    return line;
+    uint8_t macBuffer[chip::DeviceLayer::ConfigurationManager::kPrimaryMACAddressLength];
+    MutableByteSpan mac(macBuffer);
+    if (chip::DeviceLayer::ConfigurationMgr().GetPrimaryMACAddress(mac) != CHIP_NO_ERROR)
```

Comment:
i know this is the reference demo app. 
but here we obtain no matter what the primary mac address (which may be Wi-Fi).
How do we handle a device ETH controller which don't have this WOL feature ?
Does it exist an interface api to know if this capability is supported ?
Are we assuming the device manufacturer to set nicely the PICS related to this feature ?

Reply:
this is an optional cluster so a device that does not support WoL would not include this cluster. Yes, the device maker needs to make sure that they include (or don't include) this cluster. Also, they need to make sure that the Mac address used is for the correct network interface